src/config.rs

```diff
@@ -1048,6 +1101,50 @@ pub mod tests {
         validate_metadata_vector(&cfg.ca_headers, expected_ca_headers.clone());
     }
 
+    #[test]
+    fn test_ztunnel_identity_workload() {
```

Comment:
https://docs.rs/envtestkit/latest/envtestkit/, wonder if we should use something like this for the testing 🤔

Reply:
its basically broken to set env vars in tests. unless you use the same library in ALL tests (so they share a global mutex to serialize tests) which sucks.

Reply:
Better to structure in a way such that you do not depend on the env var.

Reply:
Not sure how to restructure a test which boils down to "we correctly read env vars". Maybe we just remove this test and the functionality will be stressed at a higher level

Reply:
@harsh-ps-2003 , we should probably clean this up. We don't want to be setting env vars in tests but now there logic uses battle tested parse rather than a custom env var reading anyway.